### PR TITLE
MULE-10581: Batch process cannot resume after a crash, if there is a corrupted file in the ObjectStore

### PR DESCRIPTION
**Pull request by mbuchwald**

### CODE REVIEW

**Comment by mbuchwald** [comment]
(Wait until cloudbees finishes running)

File: core/src/main/java/org/mule/util/store/PersistentObjectStorePartition.java [hunk 1]
```diff
@@ -265,9 +265,20 @@ private synchronized void loadStoredKeysAndFileNames() throws ObjectStoreExcepti
...
+                {
+                    File file = files[i];
+                    StoreValue<T> storeValue = deserialize(file);
+                    realKeyToUUIDIndex.put(storeValue.getKey(), file.getName());
+                }
+                catch (ObjectStoreException e)
+                {
+                    if (logger.isWarnEnabled())
+                    {
+                        logger.warn("Could not deserialize an ObjectStore file");
```
Comment (marianogonzalez) [inline core/src/main/java/org/mule/util/store/PersistentObjectStorePartition.java]: This should also make it clear that the data will be lost... Careful with the language you use not to freak the user out

Comment (marianogonzalez) [inline core/src/main/java/org/mule/util/store/PersistentObjectStorePartition.java]: why not moving the file to a special directory instead? if you delete it we'll never have any tools to try to diagnose or have a sense of why the file is corrupted. Also, at the customer's eyes, moving the file is less violent than deleting it

File: core/src/test/java/org/mule/util/store/PersistentObjectStorePartitionTestCase.java [hunk 2]
```diff
@@ -66,6 +68,24 @@ public void indicatesUnexistentKeyOnRetrieveError() throws ObjectStoreException
...
+        try
+        {
+            File.createTempFile("temp", ".obj", objectStoreFolder.getRoot());
+            partition.store(KEY, VALUE);
+            // Expect the new stored object, and the partition-descriptor file
+            assertEquals(2, objectStoreFolder.getRoot().listFiles().length);
+        }
+        catch (Exception e)
+        {
+            fail("Supposed to have skipped corrupted or unreadable files");
```
Comment (marianogonzalez) [inline core/src/test/java/org/mule/util/store/PersistentObjectStorePartitionTestCase.java]: you're missing an assertion that the file actually got moved

File: core/src/main/java/org/mule/util/store/PersistentObjectStorePartition.java [hunk 3]
```diff
@@ -266,8 +267,23 @@ private synchronized void loadStoredKeysAndFileNames() throws ObjectStoreExcepti
...
+                {
+                    if (logger.isWarnEnabled())
+                    {
+                        logger.warn(String.format(
+                                "Could not deserialize the ObjectStore file: %s. The file will be skipped and moved " +
+                                "to the Garbage folder", file.getName()));
+                    }
+                    String garbageFileName = file.getParentFile().getParentFile().getParent() + File.separator +
+                                             GARBAGE_FOLDER + File.separator + file.getParentFile().getName();
+                    FileUtils.moveFileToDirectory(file, new File(garbageFileName), true);
```
Comment (marianogonzalez) [inline core/src/main/java/org/mule/util/store/PersistentObjectStorePartition.java]: static imports

File: core/src/main/java/org/mule/util/store/PersistentObjectStorePartition.java [hunk 4]
```diff
@@ -266,8 +267,23 @@ private synchronized void loadStoredKeysAndFileNames() throws ObjectStoreExcepti
...
+                }
+                catch (ObjectStoreException e)
+                {
+                    if (logger.isWarnEnabled())
+                    {
+                        logger.warn(String.format(
+                                "Could not deserialize the ObjectStore file: %s. The file will be skipped and moved " +
+                                "to the Garbage folder", file.getName()));
+                    }
+                    String garbageFileName = file.getParentFile().getParentFile().getParent() + File.separator +
```
Comment (marianogonzalez) [inline core/src/main/java/org/mule/util/store/PersistentObjectStorePartition.java]: isn't it best if you just calculate the delta between the app.home and the actual file. This logic is very fragile when it comes to changes in the deployment structure

**Pull request merged**